### ISSUE TITLE
Webview2 rendering incorrectly and inconsistently

### ISSUE THREAD

**Issue by appindus-ibiller** [MicrosoftEdge/WebView2Feedback#2979]
**Description**
I've got a problem where I load a very basic HTML page showing a video feed from an external source into an embedded webview2 object in WinUI3 XAML (<WebView2 width, height, zindex Source=source />) and the rendering is inconsistent.
The above is rendered as:
<html style="height: 100%;"><head><meta name="viewport" content="width=device-width, minimum-scale=0.1"></head><body style="margin: 0px; background: #0e0e0e; height: 100%"><img style="display: block;-webkit-user-select: none;margin: auto;background-color: hsl(0, 0%, 25%);" src="source"></body></html>

It will render correctly for a while, and then it will appear off center and zoomed in. I've found deleting the EBWebView folder fixes the problem. I've also found that there's a way to clear this information out programmatically via CoreWebView2Profile.ClearBrowsingDataAsync? The WebView2.CoreWebView2.Profile property does not exist for me.

I'd love to be able to not have the bug occur (definitely not a programming issue), and/or be able to disable cached browsing data (again why is it caching incorrectly!?), and/or be able to clear it out programmatically myself.

**Version**
This is occurring on the latest WebView2 runtime download from the MS website.
I'm running .NET 6.0 with WinUI3 on WinAppSDK 1.1.5
Target machine is Windows 10 17763

**Repro Steps**
Rendered WebView2 dozens of times. Expected same result every time, got bugged cached result after a while.

**Comment by victorthoang** [comment]
Hello @appindus-ibiller,

Some general questions for you before I find a dev to look into this:

- Could you provide the steps on how we might be able to reproduce this issue?
- Also, your html snippet uses an image tag. Not sure if that is causing an issue?
- Does this happen in the Edge browser?

Thanks!
Victor

**Comment by appindus-ibiller** [comment]
@victorthoang Hi, sorry I should really update the title/desc. I figured out that this is really just a side effect of that image object being tapped or pinched. I can't seem to disable user input using the properties that would imply that they disable tap, pinch, and zoom (https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2settings?view=webview2-dotnet-1.0.1418.22). So between this and the inability to clear browsing data, this control is missing a lot of functionality. I can sort of understand not exposing the clear browsing data if its not implemented but these other properties exist and don't seem to do anything?

**Comment by victorthoang** [comment]
@appindus-ibiller thanks for providing further information.

I'll assign this to a dev to see if we can have any information for you. Will need to confirm which work stream this is.

Are you able to reproduce this pretty easily? (e.g. simply tapping/pinching the image object?*). 

Thanks,
Victor

**Comment by appindus-ibiller** [comment]
yes, exactly that. IsZoomControlEnabled, IsPinchZoomEnabled, and IsTapEnabled don't seem to change the behavior when set to false. So basically unable to prevent them from zooming, and then the next time that control hits that url its cached and I can't clear the cache either (Kiosk application, so subsequent users see the zoomed in webview and are confused).

**Comment by david-risney** [comment]
Changes to properties on CoreWebView2.Settings generally don't apply until the next document loaded in WebView2. Are you setting the property values before navigating to the page where you experience the issue? If its still not working we'll need to look into why those settings aren't working.

Otherwise, talking about work arounds: One work around might be to disable pinch zoom in the page. You might be able to use CSS touch-action: none or listen to the appropriate pointer events in JS and cancel them to prevent the pinch zoom. If you don't own the HTML content and cannot change it directly, you can try using ExecuteScript to inject the changes into the page - if its a limited set of pages you can identify by the URI.

Regarding clearing data, the CoreWebView2.Profile property isn't in the version of the SDK you're using. It doesn't look like its available in the latest version of the WinApp SDK yet either. It is available now in the WebView2 SDK for use with WinForms, WPF, and WinUI2. It should be available in a future version of the WinApp SDK.
